src/store/card.js

```diff
@@ -86,8 +87,90 @@ export default {
 						return true
 					}
 
-					return card.title.toLowerCase().includes(getters.getSearchQuery.toLowerCase())
-						|| card.description.toLowerCase().includes(getters.getSearchQuery.toLowerCase())
+					let hasMatch = true
+					const matches = getters.getSearchQuery.match(/(?:[^\s"]+|"[^"]*")+/g)
+
+					const filterOutQuotes = (q) => {
+						if (q[0] === '"' && q[q.length - 1] === '"') {
+							return q.substr(1, -1)
+						}
+						return q
+					}
+					for (const match of matches) {
+						let [filter, query] = match.indexOf(':') !== -1 ? match.split(/:(.+)/) : [null, match]
+
+						if (filter === 'title') {
+							hasMatch = hasMatch && card.title.toLowerCase().includes(filterOutQuotes(query).toLowerCase())
+						} else if (filter === 'description') {
+							hasMatch = hasMatch && card.description.toLowerCase().includes(filterOutQuotes(query).toLowerCase())
+						} else if (filter === 'list') {
+							const stack = this.getters.stackById(card.stackId)
+							if (!stack) {
+								return false
+							}
+							hasMatch = hasMatch && stack.title.toLowerCase().includes(filterOutQuotes(query).toLowerCase())
+						} else if (filter === 'tag') {
+							hasMatch = hasMatch && card.labels.findIndex((label) => label.title.toLowerCase().includes(filterOutQuotes(query).toLowerCase())) !== -1
+						} else if (filter === 'date') {
+							const datediffHour = ((new Date(card.duedate) - new Date()) / 3600 / 1000)
```

Comment:
maybe use moment for diffing dates to avoid potential troubles when crossing DST offsets (rare, so probably not cricial)

Reply:
This is for now using the same logic as the local board filters but I'll split this into a separate ticket for refining both both them.

Reply:
https://github.com/nextcloud/deck/issues/2974